src/data/addressbook.go

```diff
@@ -0,0 +1,623 @@
+// it generate a random.
+func (ab *DB) genEntropy() ([]byte, error) {
+	mn, err := bip39.NewDefaultMnemonic()
+	if err != nil {
+		return nil, err
+	}
+	e, err := bip39.EntropyFromMnemonic(mn)
+	if err != nil {
+		return nil, err
+	}
+	return e, err
+}
+
+func (ab *DB) getEntropyFromConfig() ([]byte, error) {
+	tx, err := ab.db.Begin(false)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := tx.Rollback(); err != nil {
+			logrus.Fatal(err)
+		}
+	}()
+	buck := tx.Bucket(dbConfigBkt)
+	if buck == nil {
+		return nil, errBucketEmpty
+	}
+	return buck.Get([]byte("entropy")), nil
+}
+
+// getHashFromConfig get hash from config bucket.
+func (ab *DB) getHashFromConfig() ([]byte, error) {
+	tx, err := ab.db.Begin(false)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if err := tx.Rollback(); err != nil {
+			logrus.Fatal(err)
+		}
+	}()
+	buck := tx.Bucket(dbConfigBkt)
+	if buck == nil {
+		return nil, errBucketEmpty
+	}
+
+	hash := buck.Get([]byte("hash"))
+	if hash == nil {
+		return nil, errValEmpty
+	}
+
+	return hash, nil
+}
+
+// encryptContact encrypt a contact using a password with AES-GCM.
+func (ab *DB) encryptContact(c *Contact) ([]byte, error) {
+	secType, err := ab.GetSecType()
+	if err != nil {
+		return nil, err
+	}
+	switch secType {
+	case Type1:
+		return c.MarshalBinary()
+	case Type2:
+		data, err := c.MarshalBinary()
+		if err != nil {
+			return nil, err
+		}
+		return toBase64(data), nil
+	case Type3:
+		return ab.encryptAESGCM(c)
+	}
+
+	return nil, fmt.Errorf("invalid security type")
+}
+
+// Decrypt a cipher message using a password with AES-GCM and return a Contact.
+func (ab *DB) decryptContact(cipherMsg []byte) (core.Contact, error) {
+	secType, err := ab.GetSecType()
+	if err != nil {
+		return nil, err
+	}
+	switch secType {
+	case Type1:
+		c := Contact{}
+		if err := c.UnmarshalBinary(cipherMsg); err != nil {
+			return nil, err
+		}
+
+		return &c, nil
+
+	case Type2:
+		c := Contact{}
+		data, err := fromBase64(cipherMsg)
+		if err != nil {
+			return nil, err
+		}
+		if err := c.UnmarshalBinary(data); err != nil {
+			return nil, err
+		}
+		return &c, nil
+	case Type3:
+		return ab.decryptAESGCM(cipherMsg)
+	}
+
+	return nil, errInvalidSecType
+}
+
+//
+func (ab *DB) verifyHash() error {
+	hash, err := ab.getHashFromConfig()
+	if err != nil {
+		return err
+	}
+	return bcrypt.CompareHashAndPassword(hash, ab.key)
+}
+
+// addressExists search an address in the list of contacts into the AddressBook.
+// If find the address return error, else return nil.
+func (ab *DB) addressExists(address core.StringAddress, contacts []core.Contact) error {
+	for _, v := range contacts {
+		c, ok := v.(*Contact)
+		if ok {
+			for _, addrs := range c.Address {
+				if bytes.Compare(addrs.GetValue(), address.GetValue()) == 0 &&
+					bytes.Compare(addrs.GetCoinType(), address.GetCoinType()) == 0 {
+					return fmt.Errorf("Address with value: %s  and Cointype: %s alredy exist",
```

Comment:
If you have a default errors, should be there and not create a new error. pls add this error to defaults